config/initializers/mailer.rb

```diff
@@ -0,0 +1,48 @@
+Rails.application.configure do
+  #########################################
+  # Configuration Related to Action Mailer
+  #########################################
+
+  config.action_mailer.delivery_method = :smtp
+  # We need the application frontend url to be used in our emails
+  config.action_mailer.default_url_options = { host: ENV['FRONTEND_URL'] } if ENV['FRONTEND_URL'].present?
+  # We load certain mailer templates from our database. This ensures changes to it is reflected immediately
+  config.action_mailer.perform_caching = false
+  config.action_mailer.perform_deliveries = true
+  config.action_mailer.raise_delivery_errors = true
+
+  # Config related to smtp
+  smtp_settings = {
+    domain: ENV.fetch('SMTP_DOMAIN', ''),
+    address: ENV.fetch('SMTP_ADDRESS', 'localhost'),
+    port: ENV.fetch('SMTP_PORT', 587)
+  }
+
+  if ENV.fetch('SMTP_AUTHENTICATION', '').present?
```

Comment:
Would this break the exisiting apps where they have not set SMTP_AUTHENTICATION? Also why is this under a condition?

Reply:
it only makes sense to pass authentication-related parameters to hash only when an authentication type is specified via `SMTP_AUTHENTICATION`.

in the current documentation even though we ask people to set this environment variable, the value for this is hardcoded in the production config. That's the reason why the SMTP server which has a different authentication mode than `login` isn't working